HopsanGUI/HcomHandler.cpp

```diff
@@ -9091,6 +9103,197 @@ void HcomHandler::executeEqBuiltInFunction(QString fnc_call)
     }
 }
 
+void HcomHandler::executeCutBuiltInFunction(QString functionCall)
+{
+    QStringList args = extractFunctionCallExpressionArguments(functionCall);
+    if(args.size() != 2) {
+        HCOMERR("Wrong number of arguments (should be 2)");
+        mAnsType = Undefined;
+        return;
+    }
+
+    //Find first vector
+    evaluateExpression(args[0], DataVector);
+    if (mAnsType != DataVector)
+    {
+        mAnsType = Undefined;
+        HCOMERR(QString("Variable: %1 was not found!").arg(args[0]));
+        return;
+    }
+    SharedVectorVariableT pVar1 = mAnsVector;
+
+    //Find second vector
+    evaluateExpression(args[1], DataVector);
+    if (mAnsType != DataVector)
+    {
+        HCOMERR(QString("Variable: %1 was not found!").arg(args[1]));
+        mAnsType = Undefined;
+        return;
+    }
+    SharedVectorVariableT pVar2 = mAnsVector;
+
+    if(pVar1->getDataSize() != pVar2->getDataSize()) {
+        HCOMERR("Variables must be of same length.");
+        mAnsType = Undefined;
+        return;
+    }
+
+    VariableTypeT type1 = pVar1->getVariableType();
+    VariableTypeT type2 = pVar2->getVariableType();
+    if(type2 == ComplexType) {
+        HCOMERR("Second variable must not be of complex type.");
+        mAnsType = Undefined;
+        return;
+    }
+
+    mAnsVector = mpModel->getLogDataHandler()->createOrphanVariable(pVar2->getFullVariableName()+"_cut", type1);
```

Comment:
Does not matter since temporary variable, but would not pVar1->fullname make more sense

Reply:
Of course, that was what was intended.